gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest.py.j2

```diff
@@ -0,0 +1,200 @@
+            ) -> None:
+        """Instantiate the transport.
+
+        Args:
+            host ({% if service.host %}Optional[str]{% else %}str{% endif %}):
+                {{- ' ' }}The hostname to connect to.
+            credentials (Optional[google.auth.credentials.Credentials]): The
+                authorization credentials to attach to requests. These
+                credentials identify the application to the service; if none
+                are specified, the client will attempt to ascertain the
+                credentials from the environment.
+            
+            credentials_file (Optional[str]): A file with credentials that can
+                be loaded with :func:`google.auth.load_credentials_from_file`.
+                This argument is ignored if ``channel`` is provided.
+            scopes (Optional(Sequence[str])): A list of scopes. This argument is
+                ignored if ``channel`` is provided.
+            ssl_channel_credentials (grpc.ChannelCredentials): SSL credentials
+                for grpc channel. It is ignored if ``channel`` is provided.
+            quota_project_id (Optional[str]): An optional project to use for billing
+                and quota.
+            client_info (google.api_core.gapic_v1.client_info.ClientInfo):	
+                The client info used to send a user-agent string along with	
+                API requests. If ``None``, then default info will be used.	
+                Generally, you only need to set this if you're developing	
+                your own client library.
+        """
+        super().__init__(host=host, credentials=credentials)
+        self._session = AuthorizedSession(self._credentials)
+    {%- if service.has_lro %}
+
+    @property
+    def operations_client(self) -> operations_v1.OperationsClient:
+        """Create the client designed to process long-running operations.
+
+        This property caches on the instance; repeated calls return the same
+        client.
+        """
+        # Sanity check: Only create a new client if we do not already have one.
+        if 'operations_client' not in self.__dict__:
+            from google.api_core import grpc_helpers
+            self.__dict__['operations_client'] = operations_v1.OperationsClient(
+                grpc_helpers.create_channel(
+                    self._host,
+                    credentials=self._credentials,
+                    scopes=self.AUTH_SCOPES,
+                )
+            )
+
+        # Return the client from cache.
+        return self.__dict__['operations_client']
+    {%- endif %}
+
+
+    {%- for method in service.methods.values() %}
+    {# TODO(yonmg): consider using enums #}
+        {%- set ns = namespace(http=None) %}
+        {%- for option in method.options.ListFields() %}
+            {%- if option[0].name == 'http' %}
+                {%- set ns.http = option[1] %}
+            {%- endif %}
+        {%- endfor %}
```

Comment:
This is possibly a matter of opinion, but I think it would be better to have this logic live in the `Method` class. It makes it easier to unit test.
Also, what if there's more than one 'http' option? Is it deliberate that we pick the last one?

Reply:
By 'live in the method class' do you mean adding another field to the class?

Reply:
Oh, my mistake, I didn't read line 110 carefully. Yes, there can be only one option under a specific name.
Yes, I think adding another field or accessor method would be better.
Is 'http' an options extension? If so, the more idiomatic way of accessing it is shown with resources.
Strawman example:
```python
def http_opt(self) -> Optional[str]:
    opt = self.options.Extensions[http.http_opt]
    return opt[1] if opt else None
```

If this isn't an options extension, then ignore the above example.